### PR TITLE
[MRG+2] [logformatter] 'flags' format spec backward compatibility

### LINKED ISSUE #2647: After adding request flags subclasses of logformatter that rely on 'flags' format string are broken
#2082 added flags to request but it also renamed formatting string key from flags to response_flags/request_flags
```
CRAWLEDMSG = u"Crawled (%(status)s) %(request)s (referer: %(referer)s)%(flags)s"
 +CRAWLEDMSG = u"Crawled (%(status)s) %(request)s%(request_flags)s (referer: %(referer)s)%(response_flags)s" 
```

Scrapy allows you to override logformatter and this is what I have in my project. I have logformatter looking rouhgly like this

```python
# dirbot/logf.py
from scrapy.logformatter import LogFormatter


class CustomLogFormatter(LogFormatter):
    def crawled(self, request, response, spider):
        kwargs = super(CustomLogFormatter, self).crawled(
            request, response, spider)
        kwargs['msg'] = (
            u"Crawled (%(status)s) %(request)s "
            u"(referer: %(referer)s, latency: %(latency).2f s)%(flags)s"
        )
        kwargs['args']['latency'] = response.meta.get('download_latency', 0)
        return kwargs
```

now if you enable it in settings `LOG_FORMATTER = 'dirbot.logf.CustomLogFormatter'
` and  try to run it with recent master you'll get KeyError

```
2017-03-13 14:15:26 [scrapy.extensions.telnet] DEBUG: Telnet console listening on 127.0.0.1:6023
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 851, in emit
    msg = self.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.7/logging/__init__.py", line 328, in getMessage
    msg = msg % self.args
KeyError: u'flags'
Logged from file engine.py, line 238
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 851, in emit
    msg = self.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.7/logging/__init__.py", line 328, in getMessage
    msg = msg % self.args
KeyError: u'flags'
Logged from file engine.py, line 238
2017-03-13 14:15:27 [scrapy.core.scraper] DEBUG: Scraped from <200 http://www.dmoz.org/Computers/Programming/Languages/Python/Resources/>
```

So this change that renamed `flags` to `response_flags/request_flags` seems backward incompatible.

### PR DESCRIPTION
**Pull request by pawelmhm**
pass 'flags' kwarg to logger so that it is compatible with old format of CRAWLEDMSG. fixes #2647

### CODE REVIEW

File: scrapy/logformatter.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def crawled(self, request, response, spider):
                 'request_flags' : request_flags,
                 'referer': referer_str(request),
                 'response_flags': response_flags,
+                'flags': response_flags
```
Comment (kmike) [inline scrapy/logformatter.py]: Could you please add a comment which says that `flags` alias is for backwards compatibility with Scrapy < 1.4?

File: tests/test_logformatter.py [hunk 2]
```diff
@@ -64,5 +64,32 @@ def test_scraped(self):
         assert all(isinstance(x, six.text_type) for x in lines)
         self.assertEqual(lines, [u"Scraped from <200 http://www.example.com>", u'name: \xa3'])
 
+
+class LogFormatterSubclass(LogFormatter):
+    # Formatter with format spec that is same as in Scrapy before 1.3 version.
```
Comment (kmike) [inline tests/test_logformatter.py]: I think response.flags will be released as a part of 1.4, not 1.3

Comment (pawelmhm) [inline tests/test_logformatter.py]: comments updated

**Comment by kmike** [comment]
Thanks @pawelmhm, looks good to me!

**Pull request merged**